src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -57,8 +57,34 @@ public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, stri
 
             _frameworkIdentifier = frameworkIdentifier;
             _frameworkVersion = NormalizeVersion(frameworkVersion);
-            _frameworkProfile = frameworkProfile ?? string.Empty;
+
+            IsNet5Era = (_frameworkVersion.Major >= Version5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, _frameworkIdentifier));
+
+            _frameworkProfile = profile ?? string.Empty;
```

Comment:
Why not throwing if null profile?

Reply:
that would be a breaking change, so I can't do that one. :(